File: mifosng-android/src/instrumentTest/java/com/mifos/mifosxdroid/tests/ClientSearchFragmentTest.java
package com.mifos.mifosxdroid.tests;

import androidx.test.core.app.ActivityScenario;
import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner;

import com.mifos.mifosxdroid.R;
import com.mifos.mifosxdroid.online.DashboardActivity;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
import static androidx.test.espresso.action.ViewActions.typeText;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;
import static junit.framework.TestCase.assertNotNull;

@RunWith(AndroidJUnit4ClassRunner.class)
public class ClientSearchFragmentTest {

    private ActivityScenario activityScenario;

    @Before
    public void setUp() throws Exception {
        activityScenario = ActivityScenario.launch(DashboardActivity.class);
    }

    @Test
    public void testViewsAreNotNull() {
        assertNotNull(withId(R.id.tv_search));
        assertNotNull(withId(R.id.et_search));
        assertNotNull(withId(R.id.btn_search));
    }

    @Test
    public void testViewsAreOnTheScreen() {
        onView(withId(R.id.tv_search)).check(matches(withText(R.string.search)));
        onView(withId(R.id.et_search)).check(matches(isDisplayed()));
        onView(withId(R.id.btn_search)).check(matches(withText(R.string.search)));
    }

    @Test
    public void testSearchClient() {
        String clientName = "client";
        onView(withId(R.id.et_search)).perform(typeText(clientName), closeSoftKeyboard());
        onView(withId(R.id.btn_search)).perform(click());
    }
}

File: mifosng-android/src/instrumentTest/java/com/mifos/mifosxdroid/tests/ClientSearchFragmentTest.kt
package com.mifos.mifosxdroid.tests

import android.test.suitebuilder.annotation.LargeTest
import com.mifos.mifosxdroid.R
import org.junit.After
import org.junit.Rule
import org.junit.Test

/**
 * Created by Rajan Maurya on 16/4/16.
 */
@RunWith(AndroidJUnit4::class)
@LargeTest
class ClientSearchFragmentTest {
    @Rule
    var mDashboardActivity: ActivityTestRule<DashboardActivity> =
        ActivityTestRule<DashboardActivity>(
            DashboardActivity::class.java
        )

    @Before
    fun registerIdlingResource() {
        Espresso.registerIdlingResources(
            mDashboardActivity.getActivity().getCountingIdlingResource()
        )
    }

    @Test
    fun testViewsAreOnTheScreen() {
        Espresso.onView(ViewMatchers.withId(R.id.tv_search)).check(
            ViewAssertions.matches(
                ViewMatchers.withText(
                    R.string.client_search
                )
            )
        )
        Espresso.onView(withId(R.id.et_search_by_id))
            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
        Espresso.onView(withId(R.id.bt_searchClient)).check(
            ViewAssertions.matches(
                ViewMatchers.withText(
                    R.string.search
                )
            )
        )
    }

    @Test
    @Throws(Exception::class)
    fun testSearchClient() {

        // Add Client Name In EditText
        val clientname = "client"
        Espresso.onView(withId(R.id.et_search_by_id))
            .perform(ViewActions.typeText(clientname), ViewActions.closeSoftKeyboard())

        //Search from Rest API
        Espresso.onView(withId(R.id.bt_searchClient)).perform(ViewActions.click())
    }

    /**
     * Unregister your Idling Resource so it can be garbage collected and does not leak any memory.
     */
    @After
    fun unregisterIdlingResource() {
        Espresso.unregisterIdlingResources(
            mDashboardActivity.getActivity().getCountingIdlingResource()
        )
    }
}